src/main/java/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,11 @@
+package strategy;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.Storage;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Storage storage) {
+        storage.increaseQuantity(transaction.getFruit(), transaction.getQuantity());
+    }
```

Comment:
According to the SOLID principles, each class should have a single responsibility. The `SupplyOperation` class is responsible for handling the 'supply' operation, which is correct. However, ensure that this class is only called within the context of an `OperationStrategy` and that its methods are not invoked elsewhere. The `handle` method should be the only point of interaction.